src/commands/deck.ts

```diff
@@ -340,12 +407,92 @@ export class DeckCommand extends Command {
+
+		// save to re-use when we later edit the message
+		const embeds = addNotice(content);
+		const attachment = new AttachmentBuilder(deckBuffer).setName("deck.ydk");
+
 		const end = Date.now();
-		await interaction.editReply({
-			embeds: addNotice(content),
-			// a string is interpreted as a path, to upload it as a file we need a Buffer
-			files: [new AttachmentBuilder(Buffer.from(outFile, "utf-8")).setName("deck.ydk")]
+		const response = await interaction.editReply({
+			embeds,
+			files: [attachment],
+			components: [row]
 		});
+
+		const filter = (i: ButtonInteraction): boolean => {
+			// only allow the OP to click
+			return i.user.id === interaction.user.id;
+		};
+
+		// we don't await this promise, we set up the callback and then let the method complete
+		response
+			.awaitMessageComponent({ filter, componentType: ComponentType.Button, time: 60000 })
+			.then(async i => {
+				await i.deferReply({ ephemeral: !isPublic });
+
+				// upload deck with FTP
+				const filename = `${i.id}.ydk`; // use interaction ID for unique filename
+				try {
+					await this.upload(filename, deckBuffer);
+				} catch (error) {
+					// we can't specify the type in catch{}, but we can typecast now
+					// error must be from Discord.JS or basic-ftp, which we trust to only throw Errors
+					const realError = error as Error;
+					// inform user of error
+					await i.editReply({
+						content: t`Deck upload failed! Error: ${realError.message}`,
+						components: []
+					});
+					// Remove button
+					await interaction.editReply({ embeds, files: [attachment], components: [] });
+					// log the error - we're in a promise, so throwing is wrong
+					this.logger.error(serializeCommand(interaction), error);
+					return;
+				}
+
+				// disable original button
+				// prepare row to disable button on original message
+				const disabledRow = new ActionRowBuilder<ButtonBuilder>().addComponents(
+					new ButtonBuilder()
+						.setCustomId("ftp-disabled")
+						.setLabel(t`Upload Complete`)
+						.setStyle(ButtonStyle.Success)
+						.setDisabled(true)
+				);
+				await interaction.editReply({ embeds, files: [attachment], components: [disabledRow] });
+
+				// reply in affirmation
+				await i.editReply({
+					content: t`Deck successfully uploaded to <https://ygoprodeck.com/deckbuilder/?u=https://ygoprodeck.com/discord-decks/${filename}>!`,
+					components: []
+				});
+			})
+			.catch(async (err: DiscordAPIError) => {
+				// a rejection can just mean the timeout was reached without a response
+				if (err.code === DiscordjsErrorCodes.InteractionCollectorError) {
+					// remove original button
+					interaction
+						.editReply({ embeds, files: [attachment], components: [] })
+						.catch(e => this.logger.error(serializeCommand(interaction), e));
+				} else {
+					// otherwise, though, we want to treat it as a normal error
+					this.logger.error(serializeCommand(interaction), err);
```

Comment:
So in case of a regular error, the button stays?

Reply:
...Right. right right right. I was thinking to myself "oh why do we not return here", an if passing through just sometimes kind of looks like a mistake. But in this case it's not. Will revert that behaviour.